test/test-shim.ts

```diff
@@ -70,6 +74,57 @@ test('modules already in cached do not show up in trace', t => {
   t.end();
 });
 
+function moveToEmptyTempDir(): string {
```

Comment:
suggestion: this function is great, but it leaves the responsibility of "cleanup" on its callers. My suggestion would be to make it accept a function to execute:

`function inEmptyDir(fn: Function): void  { /*setup*/; fn(); /*cleanup*/  }`.

Secondly, if there is an error thrown in `fn`, cleanup is never done. My suggestion is to use a `try`/`finally` block to ensure this doesn't happen.

Reply:
Thanks for the suggestion! That's very nicely cleaner and made me miss python context managers :'(

Made the change :+1: